### PR TITLE
Fix: Normalize evaluate error and make it more user friendly

### PR DESCRIPTION
**Pull request by sarvaje**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

File: packages/connector-jsdom/src/connector.ts [hunk 1]
```diff
@@ -460,7 +460,7 @@ export default class JSDOMConnector implements IConnector {
                 debug(`Evaluation timed out after ${this._timeout / 1000}s. Killing process and reporting an error.`);
                 this.killProcess(runner);
 
-                return reject(new Error('TIMEOUT'));
+                return reject(new Error(`The script evaluation exceeded the allotted time of ${this._timeout / 1000}s`));
```
Comment (alrra) [inline packages/connector-jsdom/src/connector.ts]: For consistency, add `.` at the end.

File: packages/utils-debugging-protocol-common/src/debugging-protocol-connector.ts [hunk 2]
```diff
@@ -974,7 +974,7 @@ export class Connector implements IConnector {
             // If this gets to 60s and it hasn't been resolved, reject the Promise.
             const asyncTimeout: NodeJS.Timer = setTimeout(
                 (() => {
-                    reject(new Error('The asynchronous expression exceeded the allotted time of 60s'));
+                    reject(new Error(`The script evaluation exceeded the allotted time of ${this._timeout / 1000}s`));
```
Comment (alrra) [inline packages/utils-debugging-protocol-common/src/debugging-protocol-connector.ts]: For consistency, add `.` at the end.

File: packages/connector-jsdom/src/connector.ts (hunk 1 above)
Comment (alrra) [reply packages/connector-jsdom/src/connector.ts]: Drop `The`.

File: packages/hint-axe/src/hint.ts [hunk 3]
```diff
@@ -117,7 +117,15 @@ export default class AxeHint implements IHint {
             try {
                 result = await context.evaluate(script);
             } catch (e) {
-                await context.report(resource, null, `Error executing script: '${e.message}'. Please try with another connector.`, null, null, Severity.warning);
+                let message: string;
+
+                if (e.message.includes('evaluation exceeded')) {
+                    message = `webhint didn't return the result fast enough`;
```
Comment (alrra) [inline packages/hint-axe/src/hint.ts]: > `'webhint'`

Comment (alrra) [reply packages/hint-axe/src/hint.ts]: > did not

File: packages/hint-axe/src/hint.ts [hunk 4]
```diff
@@ -117,7 +117,15 @@ export default class AxeHint implements IHint {
...
+                } else {
+                    message = `Error executing script: '${e.message}'`;
+                }
+
+                await context.report(resource, null, `${message}. Please try again later, or you could report an issue if this problem persists.`, null, null, Severity.warning);
```
Comment (alrra) [inline packages/hint-axe/src/hint.ts]: Drop `you could`.

File: packages/hint-no-vulnerable-javascript-libraries/src/hint.ts [hunk 5]
```diff
@@ -210,7 +210,15 @@ export default class NoVulnerableJavascriptLibrariesHint implements IHint {
             try {
                 detectedLibraries = (await context.evaluate(script)) as Array<Library>;
             } catch (e) {
-                await context.report(resource, null, `Error executing script '${e.message}'. Please try with another connector.`, null, null, Severity.warning);
+                let message: string;
+
+                if (e.message.includes('evaluation exceeded')) {
+                    message = `webhint didn't return the result fast enough`;
```
Comment (alrra) [inline packages/hint-no-vulnerable-javascript-libraries/src/hint.ts]: > `'webhint'`

**Comment by sarvaje** [comment]
Ok @alrra thanks!!

**Pull request closed without merging**